pkg/actuators/machine/actuator.go

```diff
@@ -432,6 +432,11 @@ func (a *Actuator) Update(context context.Context, cluster *clusterv1.Cluster, m
 	// Parent controller should prevent this from ever happening by calling Exists and then Create,
 	// but instance could be deleted between the two calls.
 	if existingLen == 0 {
+		if (machine.Status.Phase == nil || *machine.Status.Phase == "Provisioning" || *machine.Status.Phase == "Provisioned") && (machine.Status.LastUpdated == nil || machine.Status.LastUpdated.Add(requeueAfterSeconds*time.Second).After(time.Now())) {
```

Comment:
Actuators should not care about phases. We are going towards reducing the actuator surface as much as possible to purely cloud provider interactions (i.e remove api server interactions and spec checks like this).
Phases are upper level machine controller semantics to limit and dictate actuator actions consistently across any provider.
If anything we could use `machine.spec.providerID != nil` but I'd prefer to put the aws `DescribeInstances` requests behind a retry algorithm. I've seen this issue happening after 3 min so I believe we can have a better rate of success against "AWS Eventual Consistency" by just retrying anytime time during a reasonable period, e.g 30s.
This would produce a 30s delay for the valid case where a machine should actually go failed because exists() is returning false legitimately which I think is reasonable.

Reply:
> ...but I think my preference would be to put our aws `DescribeInstances` requests behind a retry algorithm.

How parallel is the actuator?  I didn't notice any locking to guard against simultaneous goroutines operating on a single machine.  But if we're single-threaded, we certainly don't want to lock up the whole actuator for 10s of seconds while it sorts out one momentarily inconsistent machine.

And AWS returns terminated instances for a bit before forgetting about them, so the only situation where we would be unable to *find* the instance is this post-creation window.  Can you think of a non-`Phase` way of picking that out?

Reply:
>How parallel is the actuator? 
The machine controller is single-threaded, so yeh waitRetry would block. I don't think this approach would be that bad since this should only happen sporadically and only during the initial period of life of an instance.
>Can you think of a non-Phase way of picking that out?
In your check replace `phase with `spec.providerID` https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/actuators/machine/actuator.go#L108-L119

Reply:
> I don't think this approach would be that bad since this should only happen sporadically and only during the initial period of life of an instance.

I don't think the 2k+ node folks would approve ;).

> In your check replacing `phase` with `spec.providerID`...

What about it?  We should have a provider ID immediately after creation (and we've had a successful creation in these cases).  That still doesn't mean we're sufficiently far past creation to have a good chance of being through the AWS-propagating-instance-existence-internally window.

Reply:
What I'd suggest for the requeue approach is:
 `(spec.providerID != nil && machine.Status.LastUpdated.Add(requeueAfterSeconds*time.Second).After(time.Now()))`

>I don't think the 2k+ node folks would approve ;).

heh, It'd be interesting to measure. Realistically I still doubt there'd be an unreasonable negative impact for a big scale out.

Reply:
> What I'd suggest for the requeue approach is...

Switching on `ProviderID` to limit this to just-after-creation seems strange.  I'd rather use phases like I had originally or the `MachineCreationSucceeded` condition which you can see [here][1] or some such to avoid excessive delays if folks were bumping the machine status and `LastUpdated` for other reasons.  But it's not my repo; I'm sure you have a better grasp of when `LastUpdated` could get touched.  I've rerolled to take your approach (although I've kept the nil-`LastUpdated` guard to avoid panics) with acf7a9e57 -> 09c1fe9ce.

[1]: https://storage.googleapis.com/origin-ci-test/logs/release-openshift-openshift-ansible-e2e-aws-scaleup-rhel7-4.3/255/artifacts/e2e-aws-scaleup-rhel7/must-gather/registry-svc-ci-openshift-org-ci-op-b24mfpd6-stable-sha256-64c63eedf863406fbc6c7515026f909a7221472cf70283708fb7010dd5e6139e/namespaces/openshift-machine-api/machine.openshift.io/machines/ci-op-b24mfpd6-6df53-m62pz-worker-us-east-1a-centos-sl696.yaml

Reply:
Thanks @wking, generally we want to enforce predictability and consistent behaviour/workflows across providers so avoiding to rely on specific provider conditions will ease upcoming follow ups to polish this actuator logic